modos/api.py

```diff
@@ -418,39 +419,71 @@ def stream_genomics(
             raise ValueError(f"{file_path} not found in {self.path}.")
 
         if self.htsget_endpoint:
-            match GenomicFileSuffix.from_path(Path(file_path)).name:
-                case "CRAM":
-                    endpoint_type = "/reads/"
-                case "VCF" | "BCF":
-                    endpoint_type = "/variants/"
-                case _:
-                    raise ValueError("Invalid file format.")
-
             # http://domain/s3 + bucket/modo/file.cram --> http://domain/htsget/reads/modo/file.cram
             # or               + bucket/modo/file.vcf.gz --> http://domain/htsget/variants/modo/file.vcf.gz
-            url = (
-                self.htsget_endpoint
-                + endpoint_type  # /reads/ or /variants/
-                + str(Path(*Path(file_path).parts[1:]))
-            )
-
-            gen_iter = slice_remote_genomics(
-                url=url,
+            con = HtsgetConnection(
+                self.htsget_endpoint,
+                Path(Path(file_path).parts[1]),
                 region=_region,
-                reference_filename=reference_filename,
-                output_filename=output_filename,
             )
-        else:
-            # assuming user did not change directory, filepath should be the
-            # relative path to the file.
-            # for the time being, we do not check the validity of the supplied reference_filename, or
-            # the reference given in the CRAM header (used if reference not supplied by user).
+            stream = con.to_pysam()
 
-            gen_iter = slice_genomics(
-                path=file_path,
-                region=_region,
-                reference_filename=reference_filename,
-                output_filename=output_filename,
+        else:
+            # filepath should be relative to __file__.
+            # defer validation check of the supplied reference_filename
+            # if missing, CRAM header reference is used instead
+            pysam_file = open_pysam(
+                Path(file_path), reference_filename=reference_filename
             )
-
-        return gen_iter
+            stream = (rec for rec in pysam_file)
+
+        return stream
+
+
+def build_modo_from_file(
```

Comment:
**question:** should we just make this a classmethod (`MODO.from_file()`)?

Reply:
yes! I think that is a good idea!

Reply:
Done!